lib/partitions_validator_utils.pm

```diff
@@ -24,4 +30,64 @@ sub validate_partition_table {
     assert_equals($args->{table_type}, $table_type, "Partition table type does not correspond to the expected one.");
 }
 
+sub validate_partition_table {
+    my ($args) = @_;
+    return if check_var('BACKEND', 's390x');                                       # fdisk output does not show partition table for dasd
+    record_info("Check $args->{table_type}", "Verify that partition table type is $args->{table_type}");
+    my $table_type = (split(/\"/, script_output("blkid $args->{device}")))[-1];
+    assert_equals($args->{table_type}, $table_type, "Partition table type does not correspond to the expected one.");
+}
+
+sub validate_partition_creation {
+    my ($args) = @_;
+    my @lsblk_output = split(/\n/, script_output("lsblk -n "));
+    record_info("Check $args->{mount_point}", "Verify that partition $args->{mount_point} was created.");
+    my $check = 0;
+    if ($args->{mount_point} eq 'swap') {
+        foreach (@lsblk_output) {
+            if ($_ =~ /\[SWAP\]/) {
+                $check = 1;
+            }
+            last if ($check == 1);
+        }
+    }
+    else {
+        foreach (@lsblk_output) {
+            if ($_ =~ /$args->{mount_point}\z/) {
+                $check = 1;
+            }
+            last if ($check == 1);
+        }
+    }
+    assert_equals($check, 1, "The $args->{mount_point} partition was not created.");
+}
+
+sub validate_filesystem {
+    my ($args) = @_;
+
+    record_info("Check filesystem", "Verify that $args->{mount_point} partition filesystem is $args->{fs_type}");
+    my @df_output = split(/\n/, script_output("df -T $args->{mount_point}"));
+    my $type      = ((split(/\s*\s/, $df_output[1])))[1];
+    assert_equals($args->{fs_type}, $type, "Filesystem type does not correspond to the expected one.");
+}
+
+sub validate_read_write {
+    my ($args) = @_;
+
+    assert_script_run("echo Hello > $args->{mount_point}/emptyfile", fail_message => 'Failure while writing in ' . $args->{mount_point});
+    assert_script_run("grep Hello $args->{mount_point}/emptyfile",   fail_message => 'Failure while reading from ' . $args->{mount_point});
+}
+
+sub validate_unpartitioned_space {
+    my ($args) = @_;
+
+    record_info("Check $args->{disk} partitioning", "Verify that the '$args->{disk}' does not have unpartitioned disk space.");
+    my @parted_output = split(/\n/, script_output("parted \/dev\/$args->{disk} unit GB print free"));
+    my $unpartitioned;
+    foreach (@parted_output) {
+        if ($_ =~ /(\S+)\s* Free\sSpace/) { $unpartitioned = $1 }
```

Comment:
you can use post-fix notations and avoid the curly brackets and capture groups in regex to make it easy to read what you are capturing (no need for \s between 'Free' and 'Space' and probably there is at least 1 space between 'GB' and 'Free'). Something like this:
`$unpartitioned = $1 if ($_ =~ /(\S+)\s+ Free Space;`